src/webviews/extension-side/dataviewer/kernelDataViewerDependencyImplementation.ts

```diff
@@ -0,0 +1,107 @@
+function kernelHasSession(kernel: IKernel): kernel is IKernelWithSession {
+    return Boolean(kernel.session);
+}
+
+/**
+ * Uses the Kernel to manage the dependencies of a Data Viewer.
+ */
+export class KernelDataViewerDependencyImplementation extends BaseDataViewerDependencyImplementation {
+    protected async execute(command: string, kernel: IKernelWithSession): Promise<(string | undefined)[]> {
+        const outputs = await executeSilently(kernel.session, command);
+        const error = outputs.find((item) => item.output_type === 'error');
+        if (error) {
+            traceWarning(DataScience.failedToGetVersionOfPandas(), error.message);
+        }
+        return outputs.map((item) => item.text?.toString());
+    }
+
+    protected async getVersion(kernel: IKernelWithSession): Promise<SemVer | undefined> {
+        try {
+            const outputs = await this.execute(kernelGetPandasVersion, kernel);
+            return outputs.map((text) => (text ? parseSemVer(text.toString()) : undefined)).find((item) => item);
+        } catch (e) {
+            traceWarning(DataScience.failedToGetVersionOfPandas(), e.message);
+            return;
+        }
+    }
+
+    private async installMissingDependencies(kernel: IKernelWithSession): Promise<void> {
+        sendTelemetryEvent(Telemetry.PythonModuleInstall, undefined, {
+            action: 'displayed',
+            moduleName: ProductNames.get(Product.pandas)!
+        });
+
+        const command = `${kernelPackaging(kernel)} install pandas`;
+
+        if (await this.promptInstall()) {
+            try {
+                await this.execute(command, kernel);
+                sendTelemetryEvent(Telemetry.UserInstalledPandas);
+            } catch (e) {
+                sendTelemetryEvent(Telemetry.UserInstalledPandas, undefined, undefined, e);
+                throw new Error(DataScience.failedToInstallPandas());
+            }
+        } else {
+            sendTelemetryEvent(Telemetry.UserDidNotInstallPandas);
+            throw new Error(
+                DataScience.pandasRequiredForViewing().format(pandasMinimumVersionSupportedByVariableViewer)
+            );
+        }
+    }
+
+    async checkAndInstallMissingDependencies(kernel: IKernel): Promise<void> {
+        sendTelemetryEvent(Telemetry.DataViewerUsingKernel);
+
+        if (!kernelHasSession(kernel)) {
+            sendTelemetryEvent(Telemetry.NoActiveKernelSession);
+            throw new Error(DataScience.noActiveKernelSession());
+        }
+
+        const pandasVersion = await this.getVersion(kernel);
+
+        if (pandasVersion) {
+            if (pandasVersion.compare(pandasMinimumVersionSupportedByVariableViewer) > 0) {
+                return;
+            }
+            sendTelemetryEvent(Telemetry.PandasTooOld);
+            // Warn user that we cannot start because pandas is too old.
+            const versionStr = `${pandasVersion.major}.${pandasVersion.minor}.${pandasVersion.build}`;
```

Comment:
This code can also be moved into the base class, all the implementations need to do is get the version and return it, and the base class can check if a version was available or not.